provides.py

```diff
@@ -82,13 +90,21 @@ def auth_user(self):
                              'group': conv.get_remote('auth_group')}))
         return requests
 
-    def sign_auth_request(self, scope, kubelet_token, proxy_token,
+    def sign_auth_request(self, scope, user, kubelet_token, proxy_token,
                           client_token):
         """Send authorization tokens to the requesting unit """
         conv = self.conversation(scope)
-        conv.set_remote(data={'kubelet_token': kubelet_token,
-                              'proxy_token': proxy_token,
-                              'client_token': client_token})
+        cred = {'scope': scope,
+                'kubelet_token': kubelet_token,
+                'proxy_token': proxy_token,
+                'client_token': client_token}
+        if not db.get('creds'):
+            db.set('creds', {})
+
+        all_creds = db.get('creds')
+        all_creds[user] = cred
+        db.set('creds', all_creds)
+        conv.set_remote(data={'creds': json.dumps(all_creds)})
```

Comment:
We're sending all credentials to all units. Isn't that a bit much? Better to only send kubernetes-worker-0 the credentials that it needs, not the credentials for everyone.

Reply:
This is a good point you bring up. I had it the way you suggest but there was a race condition (https://github.com/juju-solutions/interface-kube-control/commit/29c2aa1969179b7568e9ebf123d768f2e90eb5e2) . If I am not mistaken I was discussing this with @johnsca but honestly right now I do not remember much. I am confusing this interface change with the reason we reverted this: https://github.com/juju-solutions/interface-http/pull/7 . In any case, trying to reproduce this race condition right now.

Reply:
Here is what I see so far. `conv.set_remote(data={'creds': json.dumps(all_creds)})` will update the conversation data of all workers (all conversations) even though we ask for `conv = self.conversation(scope)`. In case of concurrent worker instantiations some of them may miss their token.

Reply:
Oh man, that's nuts. Well, if there's no easy way around it then I'm +1 to leaving this as is.